syntax = "proto3";

package fraud_detection;

message User {
    string name = 1;
    string contact = 2;
}

message CreditCard {
    string number = 1;
    string expirationDate = 2;
    string cvv = 3;
}

service FraudDetectionService {
    rpc CheckFraudUser (FraudDetectionRequest) returns (FraudDetectionResponse) {}
    rpc CheckFraudCreditCard (FraudDetectionRequest) returns (FraudDetectionResponse) {}
}

message FraudDetectionRequest {
    User user = 1;
    CreditCard creditCard = 2;
}

message FraudDetectionResponse {
    bool is_fraudulent = 1;
    string reason = 2;
}